.cbmc-batch/jobs/aws_cryptosdk_sign_header/aws_cryptosdk_sign_header_harness.c

```diff
@@ -28,10 +28,7 @@ void aws_cryptosdk_sign_header_harness() {
     struct aws_byte_buf header;
 
     /* assumptions*/
-    __CPROVER_assume(props);
-    __CPROVER_assume(
-        props->impl->cipher_ctor == EVP_aes_128_gcm || props->impl->cipher_ctor == EVP_aes_192_gcm ||
-        props->impl->cipher_ctor == EVP_aes_256_gcm);
+    __CPROVER_assume(aws_cryptosdk_alg_properties_is_valid(props));
```

Comment:
has this been added as a precondition on the function?

Reply:
It hadn't been but I've gone ahead and added it as a precondition for both aws_cryptosdk_sign_header and aws_cryptosdk_verify_header.